src/main/java/io/strimzi/kafka/quotas/StaticQuotaCallback.java

```diff
@@ -73,14 +76,12 @@ public Double quotaLimit(ClientQuotaType quotaType, Map<String, String> metricTa
 
         // Don't allow producing messages if we're beyond the storage limit.
         long currentStorageUsage = storageUsed.get();
-        if (ClientQuotaType.PRODUCE.equals(quotaType) && currentStorageUsage > storageQuotaSoft && currentStorageUsage < storageQuotaHard) {
-            double minThrottle = quotaMap.getOrDefault(quotaType, Quota.upperBound(Double.MAX_VALUE)).bound();
-            double limit = minThrottle * (1.0 - (1.0 * (currentStorageUsage - storageQuotaSoft) / (storageQuotaHard - storageQuotaSoft)));
+        if (ClientQuotaType.PRODUCE.equals(quotaType)) {
+            double minThrottle = quotaMap.getOrDefault(quotaType, Quota.upperBound(currentQuotaFactor)).bound();
+            final double produceQuota = minThrottle * currentQuotaFactor;
+            final double limit = Math.max(produceQuota, 1.0D);
             maybeLog(lastLoggedMessageSoftTimeMs, "Throttling producer rate because disk is beyond soft limit. Used: {}. Quota: {}", storageUsed, limit);
```

Comment:
Log message is now slightly miss leading, as the usage could be beyond the hard limit too.